drivers/net/rpmsgdrv.c

```diff
@@ -1152,6 +1153,6 @@ int net_rpmsg_drv_init(FAR const char *cpuname,
 
   /* Register the device with the OS so that socket IOCTLs can be performed */
 
-  netdev_register(dev, lltype);
-  return OK;
+  ret = netdev_register(dev, lltype);
```

Comment:
free priv and unregister rpmsg callback before return

Reply:
added little bit more error handling to init()